src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs

```diff
@@ -112,23 +112,20 @@ public static async Task<bool> CheckDefaultZipApp(string filePath)
 
 		public static IAsyncOperation<BaseStorageFolder> FromPathAsync(string path)
 		{
-			return AsyncInfo.Run<BaseStorageFolder>(async (cancellationToken) =>
+			if (!FileExtensionHelpers.IsBrowsableZipFile(path, out var ext))
 			{
-				if (!FileExtensionHelpers.IsBrowsableZipFile(path, out var ext))
-				{
-					return null;
-				}
-				var marker = path.IndexOf(ext, StringComparison.OrdinalIgnoreCase);
-				if (marker is not -1)
+				return null;
```

Comment:
They are not equivalent, instead using

```suggestion
				return Task.FromResult<BaseStorageFolder>(null).AsAsyncOperation();
```